frame/nomination-pools/src/tests.rs

```diff
@@ -5292,3 +5368,1231 @@ mod reward_counter_precision {
 			});
 	}
 }
+
+mod commission {
+	use super::*;
+
+	#[test]
+	fn set_commission_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			let pool_id = 1;
+			let root = 900;
+
+			// Commission can be set by the `root` role.
+
+			// When:
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(root),
+				pool_id,
+				Some((Perbill::from_percent(50), root))
+			));
+
+			// Then:
+			assert_eq!(
+				pool_events_since_last_call(),
+				vec![
+					Event::Created { depositor: 10, pool_id },
+					Event::Bonded { member: 10, pool_id, bonded: 10, joined: true },
+					Event::PoolCommissionUpdated {
+						pool_id,
+						current: Some((Perbill::from_percent(50), root))
+					},
+				]
+			);
+
+			// Commission can be updated only, while keeping the same payee.
+
+			// When:
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(root),
+				1,
+				Some((Perbill::from_percent(25), root))
+			));
+
+			// Then:
+			assert_eq!(
+				pool_events_since_last_call(),
+				vec![Event::PoolCommissionUpdated {
+					pool_id,
+					current: Some((Perbill::from_percent(25), root))
+				},]
+			);
+
+			// Payee can be updated only, while keeping the same commission.
+
+			// Given:
+			let payee = 901;
+
+			// When:
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(root),
+				pool_id,
+				Some((Perbill::from_percent(25), payee))
+			));
+
+			// Then:
+			assert_eq!(
+				pool_events_since_last_call(),
+				vec![Event::PoolCommissionUpdated {
+					pool_id: 1,
+					current: Some((Perbill::from_percent(25), payee))
+				},]
+			);
+
+			// Pool earns 80 points and a payout is triggered.
+
+			// Given:
+			assert_ok!(Balances::mutate_account(&default_reward_account(), |a| a.free += 80));
+			assert_eq!(
+				PoolMembers::<Runtime>::get(10).unwrap(),
+				PoolMember::<Runtime> { pool_id, points: 10, ..Default::default() }
+			);
+
+			// When:
+			assert_ok!(Pools::claim_payout(RuntimeOrigin::signed(10)));
+
+			// Then:
+			assert_eq!(
+				pool_events_since_last_call(),
+				vec![Event::PaidOut { member: 10, pool_id, payout: 60 }]
+			);
+			assert_eq!(RewardPool::<Runtime>::current_balance(pool_id), 20);
```

Comment:
Should we check that the balance of `901` was also updated accordingly with the comission?